lua/hollow/hollow.lua

```diff
@@ -0,0 +1,282 @@
+local bint = require('.bint')(256)
+local sqlite3 = require("lsqlite3")
+
+Owner = Owner or ao.env.Process.Owner
+
+Items = Items or {}
+Balances = Balances or {}
+
+ao.authorities = {"fcoN_xJeisVsPXA-trzVAuIiqO3ydLQxM-L4XbrQKzY", "jmGGoJaDYDTx4OCM7MP-7l-VLIM4ZEGCS0cHPsSmiNE"}
+
+--
+-- WRITE
+-- 1. Add-Item - add new item to the hollow or change price/qty
+-- 3. Credit-Notice - create order of the items
+-- READ
+-- 1. Read-Item - read item with specific name
+-- 2. Read-Hollow - read all the items in the hollow
+-- 3. Read-Balance - read player's items balance
+-- 4. Read-Orders - read orders from db
+--
+
+-- Open the database
+db = db or sqlite3.open_memory()
+
+-- init db
+db:exec([[
+create table if not exists orders(
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    wallet_address TEXT NOT NULL,
+	timestamp INTEGER NOT NULL,
+	item_type TEXT NOT NULL,
+	item_qty INTEGER NOT NULL,
+	payment INTEGER NOT NULL
+);
+]])
+
+Handlers.add('Credit-Notice', Handlers.utils.hasMatchingTag('Action', 'Credit-Notice'), function(msg)
```

Comment:
maybe new features from the AOS 2.0 could be used - like receive https://github.com/permaweb/aos/blob/main/process/ao.lua#L244

https://hackmd.io/@ao-docs/rk_R2S_O0#Receive-the-await-of-aos